wrapper/src/main/java/software/amazon/jdbc/exceptions/ExceptionManager.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.exceptions;
+
+public class ExceptionManager {
+  protected ExceptionHandler customHandler;
+
+  public void setCustomHandler(final ExceptionHandler exceptionHandler) {
+    this.customHandler = exceptionHandler;
+  }
+
+  public boolean isLoginException(final String driverProtocol, final Throwable throwable) {
+    if (customHandler != null) {
+      return customHandler.isLoginException(throwable);
+    }
+
+    final ExceptionHandler handler = getHandler(driverProtocol);
+    return handler.isLoginException(throwable);
+  }
+
+  public boolean isNetworkException(final String driverProtocol, final Throwable throwable) {
+    if (customHandler != null) {
+      return customHandler.isNetworkException(throwable);
+    }
+
+    final ExceptionHandler handler = getHandler(driverProtocol);
+    return handler.isNetworkException(throwable);
+  }
+
+  private ExceptionHandler getHandler(final String driverProtocol) {
+    if (driverProtocol == null) {
+      return new SimpleExceptionHandler();
+    }
+
+    switch (driverProtocol) {
+      case "jdbc:mysql://":
```

Comment:
Mysql driver supports advanced protocols like "jdbc:mysql:loadbalance://" and "jdbc:mysql:replication://". I believe other drivers may do the same. Would it be more flexible to use contains() on driverProtocol rather than switch?